lib/Minz/ModelPdo.php

```diff
@@ -68,22 +73,27 @@ private function dbConnect() {
 	}
 
 	/**
-	 * Créé la connexion à la base de données à l'aide des variables
-	 * HOST, BASE, USER et PASS définies dans le fichier de configuration
+	 * Create the connection to the database using the variables
+	 * HOST, BASE, USER and PASS variables defined in the configuration file
+	 * @param Minz_Session|null $currentUser
+	 * @param PDO|null $currentPdo
+	 * @throws Minz_ConfigurationNamespaceException
+	 * @throws Minz_PDOConnectionException
 	 */
-	public function __construct($currentUser = null, $currentPdo = null) {
+	public function __construct($currentUser = null, $currentPdo = null)
+	{
 		if ($currentUser === null) {
 			$currentUser = Minz_Session::param('currentUser');
 		}
-		if ($currentPdo != null) {
+		if ($currentPdo !== null) {
 			$this->pdo = $currentPdo;
 			return;
 		}
-		if ($currentUser == '') {
+		if ($currentUser === '') {
```

Comment:
I'm not sure if stricter matching is necessarily intended here if it was written like that.

Reply:
The strict comparison is always more qualitative than the non-strict comparison. The non-strict one has too much risk of side effects.
Look at the table in the PHP documentation
https://www.php.net/manual/en/types.comparisons.php

Reply:
Changing to `===` would mean that you believe that `false`, `0`, `''`, are valid usernames, which is not the case.

Reply:
If I set === and $currentUser is an empty string, then an exception is thrown if we don't want 0, false to be accepted, so let's make it explicit.

Reply:
changed

Reply:
Incidentally, my comment was not intended for just this one specific line but just as a representative example. They were all explicitly written to be truthy or falsy, otherwise they would've been strict already.

Reply:
;) https://stackoverflow.com/questions/2401478/why-is-faster-than-in-php

Reply:
My point isn't that I like `($currentUser == '')` better than `($currentUser === '' || $currentUser === null || $currentUser === false || $currentUser === 0)`)[1] but that it entails all that and produces different results. I.e., that `($currentUser == '')` isn't the same as `($currentUser === '')`. That's the same for probably every instance where you replaced `!=` or `==` by the stricter `!==` or `===`. It's quite unlikely that truthy and falsy were used by mistake (in this particular codebase).

[1] Though for reading I probably do since it says all that so much quicker. :-) But that's neither here nor there.